[Tutorial] Use per-SM descriptors in matmul tutorial

peterbell10 opened the pull request.

ThomasRaoux commented:
Nice! How does it affect performance?

peterbell10 commented:
It does slightly lower the utilization again from 0.285 to 0.281 compared to the previous device-side update. Or down from 0.287 with no tma update whatsoever

ThomasRaoux commented on python/tutorials/09-persistent-matmul.py:
```diff
@@ -604,7 +631,8 @@ def validate(M, N, K, dtype):
-    for K in range(args.K_range[0], args.K_range[1] + 1, args.K_step):
-        bench(K, dtype)
-    proton.finalize()
+    if 1:
```
can we make this run based on a command line?

pawelszczerbuk replied:
I don't think this is even necessary, should be OK to run this always?

pawelszczerbuk commented on python/tutorials/09-persistent-matmul.py:
```diff
@@ -535,9 +558,13 @@ def bench(K, dtype, reps=10):
-        for _ in range(reps):
-            matmul_device_tma_persistent(a, b)
-            time.sleep(0.01)
+        for tiles_per_group in [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]:
```
Do we want to go over all this each run?

chris002oak commented:
Ok

El jue, 12 de sept de 2024, 2:47 p. m., pawelszczerbuk <
***@***.***> escribió:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In python/tutorials/09-persistent-matmul.py
> <https://github.com/triton-lang/triton/pull/4682#discussion_r1757558621>:
>
> > @@ -535,9 +558,13 @@ def bench(K, dtype, reps=10):
>          for _ in range(reps):
>              matmul_tma_persistent(a, b)
>              time.sleep(0.01)
> -        for _ in range(reps):
> -            matmul_device_tma_persistent(a, b)
> -            time.sleep(0.01)
> +        for tiles_per_group in [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]:
>
> Do we want to go over all this each run?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/triton-lang/triton/pull/4682#pullrequestreview-2301381246>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKXFP3U63LFAEEBW37Y7NKDZWH4UDAVCNFSM6AAAAABN5RDFXKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMBRGM4DCMRUGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

peterbell10 commented:
Okay, I've:
1. cleaned up `if 1:`
2. renamed tiles_per_group to tiles_per_update
3. made tiles_per_update a command line argument and only run a single value
